pkg/db/db_test.go

```diff
@@ -849,6 +849,230 @@ func TestDeleteApplicationLock(t *testing.T) {
+		Version             *int64
+		ExpectedDeployments []QueuedDeployment
+	}{
+		{
+			Name:    "Write and delete",
+			Env:     envName,
+			AppName: appName,
+			Version: version(1),
+			ExpectedDeployments: []QueuedDeployment{
+				{
+					EslVersion: 2,
+					Env:        envName,
+					App:        appName,
+					Version:    nil,
+				},
+				{
+					EslVersion: 1,
+					Env:        envName,
+					App:        appName,
+					Version:    version(1),
+				},
+			},
+		},
+	}
+
+	for _, tc := range tcs {
+		tc := tc
+		t.Run(tc.Name, func(t *testing.T) {
+			t.Parallel()
+			ctx := testutil.MakeTestContext()
+
+			dbHandler := setupDB(t)
+			err := dbHandler.WithTransaction(ctx, func(ctx context.Context, transaction *sql.Tx) error {
+				err := dbHandler.DBWriteDeploymentAttempt(ctx, transaction, tc.Env, tc.AppName, tc.Version)
+				if err != nil {
+					return err
+				}
+
+				err = dbHandler.DBDeleteDeploymentAttempt(ctx, transaction, tc.Env, tc.AppName)
+				if err != nil {
+					return err
+				}
+				actual, err := dbHandler.DBSelectDeploymentAttemptHistory(ctx, transaction, tc.Env, tc.AppName, len(tc.ExpectedDeployments))
+				if err != nil {
+					return err
+				}
+
+				if diff := cmp.Diff(tc.ExpectedDeployments, actual, cmpopts.IgnoreFields(QueuedDeployment{}, "Created")); diff != "" {
+					t.Fatalf("env locks mismatch (-want, +got):\n%s", diff)
+				}
+				return nil
+			})
+			if err != nil {
+				t.Fatalf("transaction error: %v", err)
+			}
+		})
+	}
+}
+
+// setupDB returns a new DBHandler with a tmp directory every time, so tests can are completely independent
+func setupDB(t *testing.T) *DBHandler {
+	dir, err := testutil.CreateMigrationsPath(2)
+	tmpDir := t.TempDir()
+	t.Logf("directory for DB migrations: %s", dir)
+	t.Logf("tmp dir for DB data: %s", tmpDir)
+	cfg := DBConfig{
+		MigrationsPath: dir,
+		DriverName:     "sqlite3",
+		DbHost:         tmpDir,
+	}
+
+	migErr := RunDBMigrations(cfg)
+	if migErr != nil {
+		t.Fatal(migErr)
+	}
+
+	dbHandler, err := Connect(cfg)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	return dbHandler
+}
+
+func SetupRepositoryTestWithDB(t *testing.T) *DBHandler {
+	migrationsPath, err := testutil.CreateMigrationsPath(2)
+	if err != nil {
+		t.Fatalf("CreateMigrationsPath error: %v", err)
+	}
+	dbConfig := &DBConfig{
+		MigrationsPath: migrationsPath,
+		DriverName:     "sqlite3",
+	}
+
+	dir := t.TempDir()
+	remoteDir := path.Join(dir, "remote")
+	localDir := path.Join(dir, "local")
+	cmd := exec.Command("git", "init", "--bare", remoteDir)
+	err = cmd.Start()
+	if err != nil {
+		t.Fatalf("error starting %v", err)
+		return nil
+	}
+	err = cmd.Wait()
+	if err != nil {
+		t.Fatalf("error waiting %v", err)
+		return nil
+	}
+	t.Logf("test created dir: %s", localDir)
+
+	dbConfig.DbHost = dir
+
+	migErr := RunDBMigrations(*dbConfig)
+	if migErr != nil {
+		t.Fatal(migErr)
+	}
+
+	dbHandler, err := Connect(*dbConfig)
+	if err != nil {
+		t.Fatal(err)
+	}
+	return dbHandler
+}
+
+// @@@@@
```

Comment:
what is this comment?

Reply:
remnants  of a challenging merge conflict. Deleting it.